doc/how_to/custom_components/reactive_html/reactive_html_widgets.md

```diff
@@ -0,0 +1,275 @@
+with the same name as a Python parameter are run automatically when the Python parameter value is
+changed.
+
+## Select Widget
+
+Lets see how to create a basic `Select` widget.
+
+```{pyodide}
+import panel as pn
+import param
+
+pn.extension()
+
+class Select(pn.reactive.ReactiveHTML):
+
+    options = param.List(doc="Options to choose from.")
+
+    value = param.String(doc="Current selected option")
+
+    _template = """
+    <style>
+    .pn-container {height: 100%;width: 100%;position:relative;}
+    .style {border: 2px dashed lightgray;border-radius:20px}
+    </style>
+    <select id="select_el" class="pn-container style" value="${value}">
+      {% for option in options %}
+      <option id="option">{{option}}</option>
+      {% endfor %}
+    </select>
+    """
+
+    _dom_events = {'select_el': ['change']}
+
+select = Select(
+    value="B",
+    options=['A', 'B', 'C'], height=50, width=300,
+)
+pn.Column(select, select.param.value)
+```
+
+Note how we used a {% for ... %}` loop to loop over the options.
+
+In this example we inserted the options as literal `str` values via `{{option}}`.
+
+Note that the example above inserted the `options` as child objects but since they are strings we could use literals instead:
+
+```html
+<select id="select" value="${value}" style="width: ${model.width}px">
+  {% for option in options %}
+  <option id="option-{{ loop.index0 }}">{{ option }}</option>
+  {% endfor %}
+</select>
+```
+
+When using child literals we have to ensure that each `<option>` DOM node has a unique ID manually by inserting the `loop.index0` value (which would otherwise be added automatically).etc>
+
+## Drawable Canvas
+
+Next we will build a more complex widget to draw on a canvas with a
+configurable line width, color and the ability to clear and save the resulting drawing.
+
+```{pyodide}
+import panel as pn
+import param
+
+pn.extension()
+
+
+class Canvas(pn.reactive.ReactiveHTML):
+    value = param.String()
+
+    color = param.Color(default="#000000")
+    line_width = param.Number(default=1, bounds=(0.1, 10))
+
+    save = param.Event()
+    clear = param.Event()
+
+    _template = """
+    <canvas
+      id="canvas_el" style="border: 1px solid;"
+      width="${model.width}" height="${model.height}"
+      onmousedown="${script('start')}" onmousemove="${script('draw')}" onmouseup="${script('end')}"
+    ></canvas>
+    """
+
+    _scripts = {
+        "render": "state.ctx = canvas_el.getContext('2d')",
+        "start": """
+          state.start = event
+          state.ctx.beginPath()
+          state.ctx.moveTo(state.start.offsetX, state.start.offsetY)""",
+        "draw": """
+          if (state.start == null)
+            return
+          state.ctx.lineTo(event.offsetX, event.offsetY)
+          state.ctx.stroke()""",
+        "end": "delete state.start",
+        "save": """
+          data.value = canvas_el.toDataURL()
+          data.save=false""",
```

Comment:
Need confirmation from @philippjfr that its OK and good practice to use `param.Event` parameters in the way I do with `save` and `reset`. I.e. be setting `data.save=false` in the script to enable triggering/ clicking it again.

Reply:
You technically should have to do that, but I guess since the value isn't synced when it is reset to false it won't trigger the next time. We should fix this and ensure it does reset automatically.

Reply:
Fixed in https://github.com/holoviz/panel/pull/6247